internal/version/tool_latest_version_finder.go

```diff
@@ -0,0 +1,186 @@
+// Copyright 2022 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package version
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/Masterminds/semver/v3"
+	"github.com/google/go-github/v38/github"
+)
+
+const (
+	minPageSize = 5
+)
+
+type LatestVersionFinder interface {
+	PrintNewVersionAvailable(w io.Writer, v, tool, bin string) error
+	IsValidTagForTool(tag, tool string) bool
+}
+
+func NewLatestVersionFinder(re ReleaseVersionDescriber) LatestVersionFinder {
+	return &latestReleaseVersionFinder{ctx: context.Background(), r: re}
+}
+
+type latestReleaseVersionFinder struct {
+	ctx context.Context
```

Comment:
[nit] prefer to pass context as an argument, context as part of a struct should only be done to keep backward compatibility https://go.dev/blog/context-and-structs